filebeat/spooler/spooler.go

```diff
@@ -61,24 +59,32 @@ func (s *Spooler) Start() {
 func (s *Spooler) run() {
 	defer s.wg.Done()
 
-	ticker := time.NewTicker(s.config.idleTimeout / 2)
-
 	logp.Info("Starting spooler: spool_size: %v; idle_timeout: %s",
 		s.config.spoolSize, s.config.idleTimeout)
 
+	timer := time.NewTimer(s.config.idleTimeout)
+
 loop:
 	for {
 		select {
 		case <-s.exit:
-			ticker.Stop()
 			break loop
 		case event := <-s.Channel:
 			if event != nil {
 				s.queue(event)
 			}
-		case <-ticker.C:
-			s.timedFlush()
+
+			// Stop timer and drain channel. See https://golang.org/pkg/time/#Timer.Reset
+			if !timer.Stop() {
+				<-timer.C
+			}
+
+		case <-timer.C:
+			debugf("Flushing spooler because of timeout. Events flushed: %v", len(s.spool))
+			s.flush()
 		}
+
+		timer.Reset(s.config.idleTimeout)
```

Comment:
I initially thought we don't need to reset the timer after the timer event happened, but this seems to be the case. Changed.